File: dlib-sys/src/wrapper/full_object_detection.cpp
#pragma once

#include <dlib/image_processing/full_object_detection.h>

namespace wrapper {

struct FullObjectDetection {
  dlib::full_object_detection inner;

  FullObjectDetection(dlib::full_object_detection full_object_detection)
      : inner(full_object_detection) {}
};
} // namespace wrapper

File: dlib-sys/src/full_object_detection.rs
use cxx::{CxxVector, UniquePtr};

pub struct FullObjectDetection {
    pub(crate) inner: UniquePtr<crate::ffi::wrapper::FullObjectDetection>,
}

impl FullObjectDetection {
    pub fn render_face_detections(&self) -> UniquePtr<CxxVector<crate::ffi::wrapper::OverlayLine>> {
        crate::ffi::wrapper::render_face_detections(self.inner.as_ref().unwrap())
    }
}
